cluster.go

```diff
@@ -987,9 +988,30 @@ func (c *Cluster) StateSync(ctx context.Context) error {
 		}
 	}
 
+	var trustedPeers []peer.ID
+	peers, err := c.consensus.Peers(ctx)
```

Comment:
The logic here assumes the local peer is a trusted peer itself in the cluster. There is no way around this in the general case as we don't know who trusts us,  but `c.config.FollowerMode` indicates that this peer declares itself as not trusted by anyone else and therefore can skip the whole checking-for-expiration thing.